lib/px/utils/pxgraphql_extractor.lua

```diff
@@ -0,0 +1,96 @@
+---------------------------------------------
+-- PerimeterX(www.perimeterx.com) Nginx plugin
+----------------------------------------------
+
+local M = {}
+
+function M.load(px_config)
+    local _M = {}
+
+    local px_logger = require("px.utils.pxlogger").load(px_config)
+    local cjson = require("cjson")
+    local px_constants = require("px.utils.pxconstants")
+    local px_common_utils = require("px.utils.pxcommonutils")
+
+    function _M.get_operation_type(body)
+        local success, body_json  = pcall(cjson.decode, body)
+        if not success then
+            return px_constants.GRAPHQL_QUERY
+        end
+
+        if not body_json["query"] then
+            return px_constants.GRAPHQL_QUERY
+        end
+        local q = body_json["query"]
+
+        local ops = px_common_utils.split(q, " ")
+        if ops and ops == px_constants.GRAPHQL_MUTATION then
+            return px_constants.GRAPHQL_MUTATION
+        else
+            return px_constants.GRAPHQL_QUERY
+        end
+    end
+
+    function _M.get_operation_name(body)
+        local success, body_json  = pcall(cjson.decode, body)
+        if not success then
+            return px_constants.GRAPHQL_QUERY
```

Comment:
`query` is not an operation name -- it is an operation type. If we cannot find the operation name then leave it as empty string